clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -50,7 +50,7 @@
 
     // This matches URIs of formats: host:port and protocol:\\host:port
     // IPv6 is supported with [ip] pattern
-    private static final Pattern HOST_PORT_PATTERN = Pattern.compile(".*?\\[?([0-9a-zA-Z\\-.:]*)\\]?:([0-9]+)");
+    private static final Pattern HOST_PORT_PATTERN = Pattern.compile(".*?(?:(?:\\[([0-9a-zA-Z:]*)\\])|([0-9a-zA-Z\\-.]*))(?::([0-9]+))?");
```

Comment:
Are both host groups really needed? Does the one that contains ':' handle both?

We have regex in other places in the project that do similar parsing we may want to keep in sync:
- https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/utils/Utils.java#L53
- https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/cluster/BrokerEndPoint.scala#L27
- https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/cluster/EndPoint.scala#L29